blaze/expr/collections.py

```diff
@@ -397,7 +398,7 @@ class Join(Expr):
     blaze.expr.collections.Merge
     """
     __slots__ = (
-        '_hash', 'lhs', 'rhs', '_on_left', '_on_right', 'how', 'suffixes',
+        '_hash', 'lhs', 'rhs', '_on_left', '_on_right', 'how', 'suffixes'
```

Comment:
is `__weakref__` in the superclass's slots? otherwise I am not sure how this is weakreferencable